src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/NonAssociationsResource.kt

```diff
@@ -196,10 +196,71 @@ class NonAssociationsResource(
     @RequestBody
     @Validated
     createNonAssociation: CreateNonAssociationRequest,
-  ): NonAssociation =
-    eventPublishWrapper(NonAssociationDomainEventType.NON_ASSOCIATION_CREATED) {
+  ): NonAssociation {
+    return eventPublishWrapper(NonAssociationDomainEventType.NON_ASSOCIATION_CREATED) {
       nonAssociationsService.createNonAssociation(createNonAssociation)
     }
+  }
+
+  @GetMapping("/non-associations")
+  @PreAuthorize("hasRole('ROLE_READ_NON_ASSOCIATIONS')")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Get optionally-filtered non-associations, paged",
+    description = "Requires READ_NON_ASSOCIATIONS role",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "A page of Non-associations are returned",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "When pagination parameters are not valid",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the READ_NON_ASSOCIATIONS role.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getNonAssociations(
+    @Schema(
+      description = "Whether to include open non-associations or not",
+      required = false,
+      defaultValue = "true",
+      example = "false",
+    )
+    @RequestParam(required = false, defaultValue = "true")
+    includeOpen: Boolean = true,
+
+    @Schema(
+      description = "Whether to include closed non-associations or not",
+      required = false,
+      defaultValue = "true",
+      example = "false",
+    )
+    @RequestParam(required = false, defaultValue = "true")
```

Comment:
I don't know how I feel about `includeClosed` by default just for this endpoint. It's inconsistent with other endpoints which can potentially lead to confusion.

I understand that:
1. this endpoint was originally return everything
2. this will be likely used mainly by Syscon for recoinciliation

But I still think if this same option is usually false by default (e.g. only open non-associations are returned by default) we should do the same in this endpoint.

Reply:
lol how _strongly_ do you feel? i don't mind flipping this, but there's already a consumer of the endpoint…